### PR TITLE
Provide `SelectChain` implementation for parachains

### PR DESCRIPTION
**Pull request by bkchr**

### CODE REVIEW

File: consensus/src/lib.rs [hunk 1]
```diff
@@ -116,38 +130,112 @@ impl<B, E, Block, RA> LocalClient for Client<B, E, Block, RA> where
...
 
 		Ok(Box::new(s))
 	}
+
+	fn parachain_head_at(
+		&self,
+		at: &BlockId<PBlock>,
+		para_id: ParaId,
+	) -> ClientResult<Option<Vec<u8>>> {
+		self.runtime_api().parachain_status(at, para_id).map(|s| s.map(|s| s.head_data.0))
```
Comment (rphmeier) [inline consensus/src/lib.rs]: IMO querying storage is much more light-client friendly.
Alternatively we could scrape the block body.

**Comment by rphmeier** [comment]
justification for migrating to runtime API instead of raw storage query?

**Comment by bkchr** [comment]
I tried to use raw storage querys, but it seems that the parachain heads are not stored anymore under `parachains heads`. I could not find any relevant code in Polkadot.

**Comment by rphmeier** [comment]
Fair enough - can revisit later.

**Pull request merged**